listenbrainz/listenstore/timescale_utils.py

```diff
@@ -114,6 +115,16 @@ def recalculate_all_user_data():
             pass
 
 
+def unlock_cron():
+    """ Unlock the cron container """
+
+    # Unlock the cron container
+    try:
+        subprocess.run(["admin/cron_lock.py", "unlock-cron"])
```

Comment:
What will happen if we start the dumps at 4am on the 15th and then an hour later (5am according to crontab) we try and run refresh_listen_count_aggregate? From what I can tell here we'll fail to make the lock file (because it already exists) but this code won't fail due to the try/except. When the aggregate process finishes, we'll delete the dump lock.

Reply:
Also a good catch!